### PR TITLE
ci/overrides: correctly add noarch overrides

### PR DESCRIPTION
**Pull request by bgilbert**
For noarch binary packages, we need to use the `evra` key and append `.noarch` to the version.

### CODE REVIEW

File: ci/overrides.py [hunk 1]
```diff
@@ -197,11 +197,13 @@ def get_binary_packages(source_nvrs):
     accepted_in_arch = {}
     client = koji.ClientSession(KOJI_URL)
 
+    archful = lambda arch: arch != 'noarch'
```
Comment (cgwalters) [inline ci/overrides.py:197]: This is an interesting idiom - I like it.  I guess one doesn't see it in statically typed languages as the extra typing required from a nested helper function hurts readability more or something?

**Review by cgwalters — APPROVED** [review]
(Somewhat superficial) LGTM

**Review by jlebon — APPROVED** [review]
Yeah, this is an awkward wart right now with how lockfiles are set up. Ideally we should be able to use `evr` consistently even with noarch packages. Thanks for this!

File: ci/overrides.py (hunk 1 above)
Comment (bgilbert) [reply ci/overrides.py:197]: Hmm, maybe.  I've also used it in Rust with closures.

**Pull request merged**